proto/wippersnapper/signal.proto

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2020-2024 Brent Rubell, Loren Norman for Adafruit Industries
+// SPDX-License-Identifier: MIT
+syntax = "proto3";
+package wippersnapper.signal;
+import "analogio.proto";
+import "checkin.proto";
+import "digitalio.proto";
+import "ds18x20.proto";
+import "error.proto";
+import "gps.proto";
+import "i2c.proto";
+import "i2c_output.proto";
+import "nanopb.proto";
+import "pixels.proto";
+import "pwm.proto";
+import "servo.proto";
+import "uart.proto";
+
+/*
+ * BrokerToDevice
+ * The BrokerToDevice message is sent from the broker to the device.
+ * It contains a oneof payload, which is a union of all the possible
+ * messages that can be sent from the broker to a device.
+ */
+message BrokerToDevice {
+  oneof payload {
+    // digitalio.proto
+    wippersnapper.digitalio.DigitalIOAdd digitalio_add       = 10;
+    wippersnapper.digitalio.DigitalIORemove digitalio_remove = 11;
+    wippersnapper.digitalio.DigitalIOEvent digitalio_event   = 12;
```

Comment:
Maybe an accident: does a `DigitalIOEvent` ever go from broker to device? I think:
- "write"s go `b2d`
- "event"s go `d2b`
That pattern holds everywhere else, and the comment above dio event says it is d2b.